tomopy/prep/stripe.py

```diff
@@ -337,3 +348,450 @@ def remove_stripe_sf(tomo, size=5, ncore=None, nchunk=None):
+    listindex = np.arange(0.0, ncol, 1.0)
+    matindex = np.tile(listindex,(nrow,1))
+    return matindex
+
+
+def _rs_algorithm_3(sinogram, size, matindex):
+    """
+    Remove stripes using the sorting technique.
+    """
+    sinogram = np.transpose(sinogram)
+    matcomb = np.asarray(np.dstack((matindex, sinogram)))
+    matsort = np.asarray(
+        [row[row[:, 1].argsort()] for row in matcomb])
+    matsort[:, :, 1] = median_filter(matsort[:, :, 1],(size,1))
+    matsortback = np.asarray(
+        [row[row[:, 0].argsort()] for row in matsort])
+    sino_corrected = matsortback[:, :, 1]
+    return np.transpose(sino_corrected)
+
+
+def _remove_stripe_based_sorting(tomo, size):
+    matindex = _create_matindex(tomo.shape[2], tomo.shape[0])
+    for m in range(tomo.shape[1]):
+        sino = tomo[:, m, :]
+        tomo[:, m, :] = _rs_algorithm_3(sino, size, matindex)
+
+ 
+def remove_stripe_based_filtering(
+        tomo, sigma=3, size=21, ncore=None, nchunk=None):
+    """
+    Remove horizontal stripes from sinogram using Nghia Vo's
+    approach :cite:
+    Algorithm 2 in the paper. Remove stripes using the filtering technique.    
+    Parameters
+    ----------
+    tomo : ndarray
+        3D tomographic data.
+    sigma: float
+        Sigma of the Gaussian window which is used to separate 
+        the low-pass and high-pass components of the intensity
+        profiles of each column.
+    size : int
+        Window size of the median filter.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs.
+    nchunk : int, optional
+        Chunk size for each core.
+ 
+    Returns
+    -------
+    ndarray
+        Corrected 3D tomographic data.
+    """
+    arr = mproc.distribute_jobs(
+        tomo,
+        func=_remove_stripe_based_filtering,
+        args=(sigma, size),
+        axis=1,
+        ncore=ncore,
+        nchunk=nchunk)
+    return arr
+
+
+def _create_1d_window(ncol, sigma, pad):
+    window = signal.gaussian(ncol + 2 * pad, std = sigma)
+    return window
+
+
+def _create_listsign(ncol, pad):
+    listsign = np.power(-1.0,np.arange(ncol + 2 * pad))
+    return listsign
+
+ 
+def _rs_algorithm_2(sinogram, window, listsign, sigma, size, pad):
+    """
+    Remove stripes using the filtering technique.
+    """    
+    sinogram = np.transpose(sinogram)
+    sinogram2 = np.pad(sinogram,((0, 0),(pad, pad)), mode = 'reflect')
+    (_, ncol) = sinogram2.shape
+    sinosmooth = np.zeros_like(sinogram)
+    for i,sinolist in enumerate(sinogram2):        
+        sinosmooth[i] = np.real(
+            ifft(fft(sinolist*listsign)*window)*listsign)[pad:ncol-pad]
+    sinosharp = sinogram - sinosmooth
+    sinosmooth_cor = median_filter(sinosmooth,(size,1))
+    return np.transpose(sinosmooth_cor + sinosharp)
+ 
+ 
+def _remove_stripe_based_filtering(tomo, sigma, size):
+    pad = 150 # To reduce artifacts caused by FFT
+    window = _create_1d_window(tomo.shape[0], sigma, pad)
```

Comment:
pad can be an optional argument. Because for small sized datasets this value can be excessively high. Again a relative metric can be used to calculate this value based on the data size.

Reply:
I was tried to reduce as many parameters as possible.